Source/UnitTests/Common/FixedSizeQueueTest.cpp

```diff
@@ -31,3 +31,83 @@ TEST(FixedSizeQueue, Simple)
 
   EXPECT_EQ(0u, q.size());
 }
+
+TEST(FixedSizeQueue, RingBuffer)
+{
+  // Testing if queue works when used as a ring buffer
+  FixedSizeQueue<int, 5> q;
+
+  EXPECT_EQ(0u, q.size());
+
+  q.push(0);
+  q.push(1);
+  q.push(2);
+  q.push(3);
+  q.push(4);
+  q.push(5);
+
+  EXPECT_EQ(5u, q.size());
+  EXPECT_EQ(1, q.pop_front());
+
+  EXPECT_EQ(4u, q.size());
+}
+
+class TestCaseClass
+{
+public:
+  static inline int num_objects = 0;  // Static fields the reason we cannot define it locally
```

Comment:
Your test class is overly complicated. If you use member variables you don't need to reset them in the destructor. The class is re-created by googletest for each TEST_F.